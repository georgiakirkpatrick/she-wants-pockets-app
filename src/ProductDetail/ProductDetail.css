.ProductDetail > section {
    padding: 10px;
}

.ProductDetail__basic-info {
    position: relative
}

.ProductDetail__button-like {
    /* margin: 10px; */
    border-radius: 15px;
    width: 100%;
    background-color: rgb(238, 238, 238);
    color: #000;
    font-size: 16px;
    font-weight: 600;
    padding: 15px 0;
    text-align: center;
}

.ProductDetail__button-section {
    position: absolute;
    top: 10px;
    right: 10px;
    height: 50%;
    width: 30%;
    display: flex;
    flex-direction: column;
    justify-content: space-around;
    align-items: flex-end;
}

.ProductDetail__carousel {
    width: 100vw;
    overflow: hidden;
}

.ProductDetail__color-list {
    display: flex;
    flex-direction: row;
}

.ProductDetail__color {
    padding: 0 10px 0 0;
}

.ProductDetail__heart {
    margin: 0;
    padding: 5px;
    color: #000;
    width: 30px;
}

.ProductDetail__images {
    display: flex;
    flex-direction: row;
    overflow: scroll;
}

.ProductDetail__images > img {
    width: 80vw;
    padding: 0 10px 0 0;
}

.ProductDetail__info-section {
    
}

.ProductDetail__product-name {
    font-size: 20px;
}