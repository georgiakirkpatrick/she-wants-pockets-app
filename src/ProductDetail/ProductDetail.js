import React, { useState } from 'react'
import data from '../DATA'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import '../FontAwesomeIcons/FontAwesomeIcons';
import ProductDetailCarousel from '../ProductDetailCarousel/ProductDetailCarousel'
import ProductDetailSection from '../ProductDetailSection/ProductDetailSection'
import './ProductDetail.css'

const ProductDetail = props => {
    const [careOpen, setCareOpen] = useState(false)
    const [designOpen, setDesignOpen] = useState(false)
    const [fabricOpen, setFabricOpen] = useState(false)
    const [notionsOpen, setNotionsOpen] = useState(false)
    const [productionOpen, setProductionOpen] = useState(false)

    const handleOpenCare = () => {
        setCareOpen(true)
    }

    const handleCloseCare = () => {
        setCareOpen(false)
    }

    const handleOpenDesign = () => {
        setDesignOpen(true)
    }

    const handleCloseDesign = () => {
        setDesignOpen(false)
    }

    const handleOpenFabric = () => {
        setFabricOpen(true)
    }

    const handleCloseFabric = () => {
        setFabricOpen(false)
    }

    const handleOpenNotions = () => {
        setNotionsOpen(true)
    }

    const handleCloseNotions = () => {
        setNotionsOpen(false)
    }

    const handleOpenProduction = () => {
        setProductionOpen(true)
    }

    const handleCloseProduction = () => {
        setProductionOpen(false)
    }

    const selectedProductId = Number(props.routeProps.match.params.productId)

    const productInfo = data.productList.find(product => 
        product.id === selectedProductId
    )

    const allProductImages = data.productImages.filter(image =>
        image.product_id === selectedProductId
    )

    const displayAllImages = allProductImages.map(image => (
        <img
            key={image.id}
            src={image.product_image_url}
            alt={image.imgAlt}
        />
    ))

    return (
        <div className='ProductDetail'>

            <ProductDetailCarousel
                id='image-carousel'
            >
                {displayAllImages}
            </ProductDetailCarousel>

            <section className='ProductDetail__basic-info'>
                <header>
                    <h2 className='ProductDetail__product-name'>
                        {productInfo.productTitle}
                    </h2>
                </header>

                <div className='ProductDetail__button-section'>
                    <button className='ProductDetail__heart' type='button'>
                        <FontAwesomeIcon icon={['far', 'heart']} />
                    </button>
                    <div className='ProductDetail__button-like'>
                        BUY
                    </div>
                </div>

                <div className='ProductDetail__info-section'>
                    <p>
                        {productInfo.brand}
                    </p>

                    <p>
                        {productInfo.price}
                    </p>

                    <p>
                        Standard US sizes 0-12
                    </p>

                    <p>
                        Fibers: 100% silk
                    </p>

                    <p>
                        Color options:
                    </p>
                </div>

                <div className='ProductDetail__color-list'>
                    <div className='ProductDetail__color'>
                        <img 
                            src='https://media.sezane.com/image/upload/c_fill,d_placeholder_dark.png,fl_progressive:semi,h_60,q_auto:best,w_43/avhc13dzwkv0pzgbmf7g.jpg'
                            alt='Black'
                        />
                    </div>

                    <div className='ProductDetail__color'>
                        <img 
                            src='https://media.sezane.com/image/upload/c_fill,d_placeholder_dark.png,fl_progressive:semi,h_60,q_auto:best,w_43/iwjijdgeeru16c6btswn.jpg'
                            alt='Paris est une fete print'
                        />
                    </div>
                </div>
            </section>
            
            <ProductDetailSection
                id='product-detail-design'
                sectionTitle='Design details'
                sectionOpen={designOpen}
                handleSectionClose={handleCloseDesign}
                handleSectionOpen={handleOpenDesign}
            >
                <p>
                    Neckline: v-neck
                </p>

                <p>
                    Sleeve length: three-quarter-length sleeves
                </p>

                <p>
                    Office-friendly: yes
                </p>

                <p>
                    Length: below the knee
                </p>

                <p>
                    Dress style: wrap dress
                </p>
            </ProductDetailSection>

            <ProductDetailSection
                id='product-detail-care'
                sectionTitle='Care instructions'
                sectionOpen={careOpen}
                handleSectionClose={handleCloseCare}
                handleSectionOpen={handleOpenCare}
            >
                <p>
                    Wash: hand wash in cold water
                </p>

                <p>
                    Dry: not provided 
                </p>
            </ProductDetailSection>

            <ProductDetailSection
                id='product-detail-production'
                sectionTitle='Production'
                sectionOpen={productionOpen}
                handleSectionClose={handleCloseProduction}
                handleSectionOpen={handleOpenProduction}
            >
                <h4>
                    Garment sewing and finishing
                </h4>

                <p>
                    Country: China
                </p>

                <p>
                    Factory: not disclosed
                </p>

                <h4>
                    Fabric cutting
                </h4>

                <p>
                    Country: not disclosed
                </p>

                <p>
                    Factory: not disclosed
                </p>

                <p>
                    Product certifications: OEKO-TEX Standard 100
                </p>

                <p>
                    Production notes: OEKO-TEX certification number CQ 1217/1IFTH
                </p>
            </ProductDetailSection>
            
            <ProductDetailSection
                sectionTitle='Fabric'
                sectionOpen={fabricOpen}
                handleSectionClose={handleCloseFabric}
                handleSectionOpen={handleOpenFabric}
            >
                <ProductDetailCarousel
                    id='section-carousel'
                >
                    <div className='ProductDetail__carousel-item'>
                        <h4>
                            Primary fabric
                        </h4>

                        <p>
                            Fabric type: not listed
                        </p>

                        <h5>
                            Dyeing and finishing
                        </h5>

                        <p>
                            Country: 
                        </p>

                        <p>
                            Factory: 
                        </p>

                        <h5>
                            Fabric mill
                        </h5>

                        <p>
                            Country: 
                        </p>

                        <p>
                            Factory: 
                        </p>

                        <p>
                            Fabric certifications: none listed
                        </p>

                        <h5>
                            Fiber(s)
                        </h5>

                        <h6>
                            Silk
                        </h6>

                        <p>
                            Fiber origin: not disclosed
                        </p>

                        <p>
                            Fiber producer: not disclosed
                        </p>

                        <p>
                            Fiber certifications: none listed
                        </p>
                    </div>

                    <div className='ProductDetail__carousel-item'>
                        <h4>
                            Secondary fabric
                        </h4>

                        <p>
                            Fabric type: not listed
                        </p>

                        <h5>
                            Dyeing and finishing
                        </h5>

                        <p>
                            Country: China
                        </p>

                        <p>
                            Factory: not disclosed
                        </p>

                        <h5>
                            Fabric mill
                        </h5>

                        <p>
                            Country: not disclosed
                        </p>

                        <p>
                            Factory: not disclosed
                        </p>

                        <p>
                            Fabric certifications: none listed
                        </p>

                        <h5>
                            Fiber(s)
                        </h5>

                        <h6>
                            Silk
                        </h6>

                        <p>
                            Fiber origin: not disclosed
                        </p>

                        <p>
                            Fiber producer: not disclosed
                        </p>

                        <p>
                            Fiber certifications: none listed
                        </p>
                    </div>
                </ProductDetailCarousel>
            </ProductDetailSection>

            <ProductDetailSection
                sectionTitle='Notions'
                sectionOpen={notionsOpen}
                handleSectionClose={handleCloseNotions}
                handleSectionOpen={handleOpenNotions}
            >
                <ProductDetailCarousel
                    id='section-carousel'
                >
                    <div className='ProductDetail__carousel-item'>
                        <h4>
                            Primary fabric
                        </h4>

                        <p>
                            Fabric type: not listed
                        </p>

                        <h5>
                            Dyeing and finishing
                        </h5>

                        <p>
                            Country: 
                        </p>

                        <p>
                            Factory: 
                        </p>

                        <h5>
                            Fabric mill
                        </h5>

                        <p>
                            Country: 
                        </p>

                        <p>
                            Factory: 
                        </p>

                        <p>
                            Fabric certifications: none listed
                        </p>
                    </div>

                    <div className='ProductDetail__carousel-item'>
                        <h4>
                            Primary fabric
                        </h4>

                        <p>
                            Fabric type: not listed
                        </p>

                        <h5>
                            Dyeing and finishing
                        </h5>

                        <p>
                            Country: 
                        </p>

                        <p>
                            Factory: 
                        </p>

                        <h5>
                            Fabric mill
                        </h5>

                        <p>
                            Country: 
                        </p>

                        <p>
                            Factory: 
                        </p>

                        <p>
                            Fabric certifications: none listed
                        </p>
                    </div>
                </ProductDetailCarousel>
            </ProductDetailSection>
        </div>
    )
}

ProductDetail.defaultProps = {
    routeProps: {
        match: {
            params: {
                productId: 4
            }
        }
    },
    productInfo: {
        brand: "Sezane",
        category: "Dresses",
        categoryId: 4,
        id: 1,
        imgAlt: "Sezane Dahlia Dress",
        pathToImg: "https://media.sezane.com/image/upload/c_fill,d_placeholder_dark.png,fl_progressive:semi,h_816,q_auto:best,w_582/avhc13dzwkv0pzgbmf7g.jpg",
        price: 200,
        productTitle: "Dahlia Dress"
    }
}

export default ProductDetail