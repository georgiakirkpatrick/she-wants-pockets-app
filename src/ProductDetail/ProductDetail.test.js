import React from 'react';
import ReactDOM from 'react-dom';
import ProductDetail from './ProductDetail';
import renderer from 'react-test-renderer';

describe ('ProductDetail', () => {
    it ('renders without crashing', () => {
        const div = document.createElement('div')
        ReactDOM.render(<ProductDetail />, div)
        ReactDOM.unmountComponentAtNode(div)
    })

    it('renders the UI as expected', () => {
        const tree = renderer
            .create(<ProductDetail />)
            .toJSON()
        expect(tree).toMatchSnapshot()    
    })
})