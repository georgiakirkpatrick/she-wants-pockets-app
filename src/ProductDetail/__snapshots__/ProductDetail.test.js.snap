// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductDetail renders the UI as expected 1`] = `
<div
  className="ProductDetail"
>
  <div
    className="ProductDetailCarousel"
    id="image-carousel"
  >
    <div
      className="ProductDetailCarousel__items"
    />
  </div>
  <section
    className="ProductDetail__basic-info"
  >
    <header>
      <h2
        className="ProductDetail__product-name"
      >
        Dahlia Dress
      </h2>
    </header>
    <div
      className="ProductDetail__button-section"
    >
      <button
        className="ProductDetail__heart"
        type="button"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-heart fa-w-16 "
          data-icon="heart"
          data-prefix="far"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M458.4 64.3C400.6 15.7 311.3 23 256 79.3 200.7 23 111.4 15.6 53.6 64.3-21.6 127.6-10.6 230.8 43 285.5l175.4 178.7c10 10.2 23.4 15.9 37.6 15.9 14.3 0 27.6-5.6 37.6-15.8L469 285.6c53.5-54.7 64.7-157.9-10.6-221.3zm-23.6 187.5L259.4 430.5c-2.4 2.4-4.4 2.4-6.8 0L77.2 251.8c-36.5-37.2-43.9-107.6 7.3-150.7 38.9-32.7 98.9-27.8 136.5 10.5l35 35.7 35-35.7c37.8-38.5 97.8-43.2 136.5-10.6 51.1 43.1 43.5 113.9 7.3 150.8z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </button>
      <div
        className="ProductDetail__button-like"
      >
        BUY
      </div>
    </div>
    <div
      className="ProductDetail__info-section"
    >
      <p>
        Sezane
      </p>
      <p>
        200
      </p>
      <p>
        Standard US sizes 0-12
      </p>
      <p>
        Fibers: 100% silk
      </p>
      <p>
        Color options:
      </p>
    </div>
    <div
      className="ProductDetail__color-list"
    >
      <div
        className="ProductDetail__color"
      >
        <img
          alt="Black"
          src="https://media.sezane.com/image/upload/c_fill,d_placeholder_dark.png,fl_progressive:semi,h_60,q_auto:best,w_43/avhc13dzwkv0pzgbmf7g.jpg"
        />
      </div>
      <div
        className="ProductDetail__color"
      >
        <img
          alt="Paris est une fete print"
          src="https://media.sezane.com/image/upload/c_fill,d_placeholder_dark.png,fl_progressive:semi,h_60,q_auto:best,w_43/iwjijdgeeru16c6btswn.jpg"
        />
      </div>
    </div>
  </section>
  <hr />
  <section>
    <button
      className="ProductDetailSection__button"
      onClick={[Function]}
      type="button"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-angle-down fa-w-10 "
        data-icon="angle-down"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 320 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M143 352.3L7 216.3c-9.4-9.4-9.4-24.6 0-33.9l22.6-22.6c9.4-9.4 24.6-9.4 33.9 0l96.4 96.4 96.4-96.4c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9l-136 136c-9.2 9.4-24.4 9.4-33.8 0z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </button>
    <header
      onClick={[Function]}
    >
      <h3>
        Design details
      </h3>
    </header>
    <div
      className="ProductDetailSection"
    >
      <p>
        Neckline: v-neck
      </p>
      <p>
        Sleeve length: three-quarter-length sleeves
      </p>
      <p>
        Office-friendly: yes
      </p>
      <p>
        Length: below the knee
      </p>
      <p>
        Dress style: wrap dress
      </p>
    </div>
  </section>
  <hr />
  <section>
    <button
      className="ProductDetailSection__button"
      onClick={[Function]}
      type="button"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-angle-down fa-w-10 "
        data-icon="angle-down"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 320 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M143 352.3L7 216.3c-9.4-9.4-9.4-24.6 0-33.9l22.6-22.6c9.4-9.4 24.6-9.4 33.9 0l96.4 96.4 96.4-96.4c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9l-136 136c-9.2 9.4-24.4 9.4-33.8 0z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </button>
    <header
      onClick={[Function]}
    >
      <h3>
        Care instructions
      </h3>
    </header>
    <div
      className="ProductDetailSection"
    >
      <p>
        Wash: hand wash in cold water
      </p>
      <p>
        Dry: not provided
      </p>
    </div>
  </section>
  <hr />
  <section>
    <button
      className="ProductDetailSection__button"
      onClick={[Function]}
      type="button"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-angle-down fa-w-10 "
        data-icon="angle-down"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 320 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M143 352.3L7 216.3c-9.4-9.4-9.4-24.6 0-33.9l22.6-22.6c9.4-9.4 24.6-9.4 33.9 0l96.4 96.4 96.4-96.4c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9l-136 136c-9.2 9.4-24.4 9.4-33.8 0z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </button>
    <header
      onClick={[Function]}
    >
      <h3>
        Production
      </h3>
    </header>
    <div
      className="ProductDetailSection"
    >
      <h4>
        Garment sewing and finishing
      </h4>
      <p>
        Country: China
      </p>
      <p>
        Factory: not disclosed
      </p>
      <h4>
        Fabric cutting
      </h4>
      <p>
        Country: not disclosed
      </p>
      <p>
        Factory: not disclosed
      </p>
      <p>
        Product certifications: OEKO-TEX Standard 100
      </p>
      <p>
        Production notes: OEKO-TEX certification number CQ 1217/1IFTH
      </p>
    </div>
  </section>
  <hr />
  <section>
    <button
      className="ProductDetailSection__button"
      onClick={[Function]}
      type="button"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-angle-down fa-w-10 "
        data-icon="angle-down"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 320 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M143 352.3L7 216.3c-9.4-9.4-9.4-24.6 0-33.9l22.6-22.6c9.4-9.4 24.6-9.4 33.9 0l96.4 96.4 96.4-96.4c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9l-136 136c-9.2 9.4-24.4 9.4-33.8 0z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </button>
    <header
      onClick={[Function]}
    >
      <h3>
        Fabric
      </h3>
    </header>
    <div
      className="ProductDetailSection"
    >
      <div
        className="ProductDetailCarousel"
        id="section-carousel"
      >
        <div
          className="ProductDetailCarousel__items"
        >
          <div
            className="ProductDetail__carousel-item"
          >
            <h4>
              Primary fabric
            </h4>
            <p>
              Fabric type: not listed
            </p>
            <h5>
              Dyeing and finishing
            </h5>
            <p>
              Country:
            </p>
            <p>
              Factory:
            </p>
            <h5>
              Fabric mill
            </h5>
            <p>
              Country:
            </p>
            <p>
              Factory:
            </p>
            <p>
              Fabric certifications: none listed
            </p>
            <h5>
              Fiber(s)
            </h5>
            <h6>
              Silk
            </h6>
            <p>
              Fiber origin: not disclosed
            </p>
            <p>
              Fiber producer: not disclosed
            </p>
            <p>
              Fiber certifications: none listed
            </p>
          </div>
          <div
            className="ProductDetail__carousel-item"
          >
            <h4>
              Secondary fabric
            </h4>
            <p>
              Fabric type: not listed
            </p>
            <h5>
              Dyeing and finishing
            </h5>
            <p>
              Country: China
            </p>
            <p>
              Factory: not disclosed
            </p>
            <h5>
              Fabric mill
            </h5>
            <p>
              Country: not disclosed
            </p>
            <p>
              Factory: not disclosed
            </p>
            <p>
              Fabric certifications: none listed
            </p>
            <h5>
              Fiber(s)
            </h5>
            <h6>
              Silk
            </h6>
            <p>
              Fiber origin: not disclosed
            </p>
            <p>
              Fiber producer: not disclosed
            </p>
            <p>
              Fiber certifications: none listed
            </p>
          </div>
        </div>
      </div>
    </div>
  </section>
  <hr />
  <section>
    <button
      className="ProductDetailSection__button"
      onClick={[Function]}
      type="button"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-angle-down fa-w-10 "
        data-icon="angle-down"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 320 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M143 352.3L7 216.3c-9.4-9.4-9.4-24.6 0-33.9l22.6-22.6c9.4-9.4 24.6-9.4 33.9 0l96.4 96.4 96.4-96.4c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9l-136 136c-9.2 9.4-24.4 9.4-33.8 0z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </button>
    <header
      onClick={[Function]}
    >
      <h3>
        Notions
      </h3>
    </header>
    <div
      className="ProductDetailSection"
    >
      <div
        className="ProductDetailCarousel"
        id="section-carousel"
      >
        <div
          className="ProductDetailCarousel__items"
        >
          <div
            className="ProductDetail__carousel-item"
          >
            <h4>
              Primary fabric
            </h4>
            <p>
              Fabric type: not listed
            </p>
            <h5>
              Dyeing and finishing
            </h5>
            <p>
              Country:
            </p>
            <p>
              Factory:
            </p>
            <h5>
              Fabric mill
            </h5>
            <p>
              Country:
            </p>
            <p>
              Factory:
            </p>
            <p>
              Fabric certifications: none listed
            </p>
          </div>
          <div
            className="ProductDetail__carousel-item"
          >
            <h4>
              Primary fabric
            </h4>
            <p>
              Fabric type: not listed
            </p>
            <h5>
              Dyeing and finishing
            </h5>
            <p>
              Country:
            </p>
            <p>
              Factory:
            </p>
            <h5>
              Fabric mill
            </h5>
            <p>
              Country:
            </p>
            <p>
              Factory:
            </p>
            <p>
              Fabric certifications: none listed
            </p>
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;
